DMCompiler/Compiler/DM/DMParser.cs

```diff
@@ -1377,6 +1382,9 @@ public partial class DMParser : Parser<Token> {
 
                         break;
                     }
+                    if (expression is DMASTAssign) {
+                        DMCompiler.Emit(WarningCode.AssignmentInConditional, Current().Location, "Assignment in conditional");
```

Comment:
Not sure if this is necessary, OD500 already errors due to the `if()` in the switch being non-constant
![image](https://github.com/OpenDreamProject/OpenDream/assets/41448081/eb6e11b5-0537-491d-a9fa-7c353ada0020)

Reply:
It's not necessary. That error is the more important thing in that case I'd say.

Reply:
Removed